store/store.go

```diff
@@ -302,6 +302,50 @@ func (s *Store[H]) HasAt(_ context.Context, height uint64) bool {
 	return height != uint64(0) && s.Height() >= height
 }
 
+// DeleteRange implements [header.Store] interface.
+func (s *Store[H]) DeleteRange(ctx context.Context, from, to uint64) error {
+	if from >= to {
+		return fmt.Errorf("header/store: invalid range(%d,%d)", from, to)
+	}
+
+	batch, err := s.ds.Batch(ctx)
+	if err != nil {
+		return fmt.Errorf("header/store: batch: %w", err)
+	}
+
+	hashes := make([]header.Hash, 0, to-from)
+	for h := from; h < to; h++ {
+		hash, err := s.heightIndex.HashByHeight(ctx, h)
+		if err != nil {
+			if errors.Is(err, datastore.ErrNotFound) {
+				log.Debugw("removing non-existent header", "height", h)
+				continue
+			}
+			return fmt.Errorf("header/store: hash by height (%d): %w", h, err)
+		}
+
+		hashes = append(hashes, hash)
+
+		if err := batch.Delete(ctx, hashKey(hash)); err != nil {
+			return fmt.Errorf("header/store: adding hash to batch: %w", err)
+		}
+		if err := batch.Delete(ctx, heightKey(h)); err != nil {
+			return fmt.Errorf("header/store: adding height to batch: %w", err)
+		}
+	}
+
+	if err := batch.Commit(ctx); err != nil {
+		return fmt.Errorf("header/store: batch commit: %w", err)
+	}
+
+	for _, h := range hashes {
+		s.cache.Remove(h.String())
```

Comment:
And heightIndex cache?

Reply:
My bad. Forgot to push next commit. Thanks for the quick comment.